src/main/java/com/github/messenger4j/quickstart/boot/MessengerPlatformCallbackHandler.java

```diff
@@ -73,508 +64,437 @@
+
+		final ReceiptTemplate receiptTemplate = ReceiptTemplate
+				.create("Peter Chang", uniqueReceiptId, "Visa 1234", "USD", Summary.create(626.66f, of(698.99f), of(57.67f), of(20.00f)),
+						of(Address.create("1 Hacker Way", "Menlo Park", "94025", "CA", "US")), of(items),
+						of(Arrays.asList(Adjustment.create("New Customer Discount", -50f), Adjustment.create("$100 Off Coupon", -100f))),
+						of("The Boring Company"), of(new URL("https://www.boringcompany.com/")), of(true), of(Instant.ofEpochMilli(1428444852L)));
+
+		final TemplateMessage templateMessage = TemplateMessage.create(receiptTemplate);
+		final MessagePayload messagePayload = MessagePayload.create(recipientId, templateMessage);
+		this.messenger.send(messagePayload);
+	}
+
+	private void sendQuickReply(String recipientId) throws MessengerApiException, MessengerIOException {
+		List<QuickReply> quickReplies = new ArrayList<>();
+
+		quickReplies.add(TextQuickReply.create("Action", "DEVELOPER_DEFINED_PAYLOAD_FOR_PICKING_ACTION"));
+		quickReplies.add(TextQuickReply.create("Comedy", "DEVELOPER_DEFINED_PAYLOAD_FOR_PICKING_COMEDY"));
+		quickReplies.add(TextQuickReply.create("Drama", "DEVELOPER_DEFINED_PAYLOAD_FOR_PICKING_DRAMA"));
+		quickReplies.add(LocationQuickReply.create());
+
+		TextMessage message = TextMessage.create("What's your favorite movie genre?", of(quickReplies), empty());
+		messenger.send(MessagePayload.create(recipientId, message));
+	}
+
+	private void sendReadReceipt(String recipientId) throws MessengerApiException, MessengerIOException {
+		this.messenger.send(SenderActionPayload.create(recipientId, SenderAction.MARK_SEEN));
+	}
+
+	private void sendTypingOn(String recipientId) throws MessengerApiException, MessengerIOException {
+		this.messenger.send(SenderActionPayload.create(recipientId, SenderAction.TYPING_ON));
+	}
+
+	private void sendTypingOff(String recipientId) throws MessengerApiException, MessengerIOException {
+		this.messenger.send(SenderActionPayload.create(recipientId, SenderAction.TYPING_OFF));
+	}
+
+	private void sendAccountLinking(String recipientId) throws MessengerApiException, MessengerIOException, MalformedURLException {
+		// Mandatory https
+		final LogInButton buttonIn = LogInButton.create(new URL("https://<YOUR_REST_CALLBACK_URL>"));
+		final LogOutButton buttonOut = LogOutButton.create();
+
+		final List<Button> buttons = Arrays.asList(buttonIn, buttonOut);
+		final ButtonTemplate buttonTemplate = ButtonTemplate.create("Log in to see an account linking callback", buttons);
+
+		final TemplateMessage templateMessage = TemplateMessage.create(buttonTemplate);
+		final MessagePayload messagePayload = MessagePayload.create(recipientId, templateMessage);
+		this.messenger.send(messagePayload);
+	}
+
+	private void handleAttachmentMessageEvent(AttachmentMessageEvent event) {
+		logger.debug("Handling QuickReplyMessageEvent");
+		for (Attachment attachment : event.attachments()) {
+			final String senderId = event.senderId();
```

Comment:
I think this should be outside of the loop.

Reply:
You are right. Solved.